src/Core/Managed/Shared/OperationTelemetryExtensions.cs

```diff
@@ -16,9 +17,13 @@ public static class OperationTelemetryExtensions
         /// </summary>
         /// <param name="telemetry">Telemetry item object that calls this extension method.</param>
         public static void Start(this OperationTelemetry telemetry)
-        {
-            var startTime = DateTimeOffset.UtcNow;
-            telemetry.Timestamp = startTime;
+        {            
+            telemetry.Timestamp = DateTimeOffset.UtcNow;
+
+            // Begin time is used internally for calculating duration of operation at the end call,
+            // and hence is stored using higher precision Clock.
+            // Stopwatch.GetTimestamp() is used (instead of ElapsedTicks) as it is thread-safe.
+            telemetry.BeginTimeInTicks = Stopwatch.GetTimestamp();
```

Comment:
I doubt the property read would have been an issue but it doesn't matter because I like this simplified version that doesn't need us to keep an instance better anyway.